torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -86,6 +86,18 @@
     }                                                                         \
   } while (0)
 
+// Macro to throw on a non-successful NCCL return value for NONBLOCKING calls.
+#define C10D_NCCL_CHECK_NONBLOCKING(cmd, failureReason)                       \
```

Comment:
Defining a new macro is okay. A neater way would be just to reuse the `C10D_NCCL_CHECK` macro: in it, rely on `#ifdef NCCL_HAS_NONBLOCKING` to add the `ncclInProgress` check.

Reply:
After a second pass after the eagerMode change, feel it is better to use a separate macro to make the logic in Macro simple: e.g., no relative long verdict statement like (if (use_non_blocking && result!= success && result !=inprogress) || (!use_non_blocking && result!= success) for better readability

Reply:
Hmm, I don't think we'd need to check `use_non_blocking` here. We just check for `result != ncclSuccess && result != ncclInProgress`. As in, after NCCL 2.14, we allow both values as immediate return. But, personal preference only, no need to take it seriously.